File: 김채연/10828.py
import sys
input = sys.stdin.readline
n = int(input())
stack = []
for _ in range(n):
    com = list(input().split())
    command = com[0]
    if command == 'push':
        stack.append(com[1])
    elif command == 'pop':
        if len(stack)!=0:
            print(stack.pop())
        else:
            print(-1)
    elif command == 'size':
        print(len(stack))
    elif command == 'empty':
        if len(stack)!=0:
            print(0)
        else:
            print(1)
    elif command == 'top':
        if len(stack)!=0:
            print(stack[len(stack)-1])
        else:
            print(-1)

File: 최시은/2024-03/10828.cpp
#include <iostream>
#include <string>
#include <vector>
#include <sstream>
using namespace std;

int main() {

    int n;
    cin >> n;
    cin.ignore();

    vector<int> stack;

    for (int i = 0; i < n; i++) {
        string input;
        getline(cin, input);

        istringstream ss(input);
        string stringBuffer;
        vector<string> command;

        while (getline(ss, stringBuffer, ' ')) {
            command.push_back(stringBuffer);
        }

        if (!command[0].compare("push")) {
            stack.push_back(stoi(command[1]));
        } else if (!command[0].compare("pop")) {
            if (stack.empty()) {
                cout << -1 << endl;
            } else {
                cout << stack.back() << endl;
                stack.pop_back();
            }
        } else if (!command[0].compare("size")) {
            cout << stack.size() << endl;
        } else if (!command[0].compare("top")) {
            if (stack.empty()) {
                cout << -1 << endl;
            } else {
                cout << stack.back() << endl;
            }
        } else if (!command[0].compare("empty")) {
            if (stack.empty()) {
                cout << 1 << endl;
            } else {
                cout << 0 << endl;
            }
        }
    }

    return 0;
}